### ISSUE TITLE
Additional postCSS plugins

### ISSUE THREAD

**Issue by faceleg** [SC5/sc5-styleguide#825]
I use a few postCSS plugins in my workflow. Is there a way to have sc5-styleguide use them as well?

When I attempt to run the given example in my gulpfile, I get sytnax errors due to the additional syntax my postCSS plugins allow.

**Comment by varya** [comment]
Could you provide the repo please?
My suggestion is that you should use `parsers: { 'postcss': 'postcss' }` option from here https://github.com/SC5/sc5-styleguide#build-options but maybe something else, cannot guess without seeing your code.

**Comment by karolis-adapt** [comment]
@varya Using CLI --parsers argument with option "{ css: 'postcss' }" and "{ postcss: 'postcss' }", the script reports the same error both times when using postcss-nested:
```
Parsing error: Please check validity of the block starting from line #17

15 |   color: inherit;
17*|   &::before {
18 |     display: inline-block;
19 |     width: 1.8em;

Syntax: css
Gonzales PE version: 3.0.0-31
```

**Comment by karolis-adapt** [comment]
[Here's a repo with code to reproduce the issue](https://github.com/karolis-adapt/sc5-postcss-testcase)

**Comment by karolis-adapt** [comment]
Oh and one more thing: currently I'm using the styleguide:ignore:start and styleguide:ignore:end flags in source css, but it would be awesome to get away without the ignore flags.

**Comment by hannu** [comment]
Currently postCSS support does not support additional plug-ins. The postCSS tree processing is located in https://github.com/SC5/sc5-styleguide/blob/master/lib/modules/parsers/postcss.js and currently it is not  possible to inject 3rd party postCSS processers

**Comment by glebmachine** [comment]
Just plug your postcss in pipes after styleguide

**Comment by karolis-adapt** [comment]
@glebmachine yep, that's the option I have resorted to now, but it is a workaround. It would be cool to have postcss additional plugin support. As hannu said, I think it should be considered for implementation.

**Comment by glebmachine** [comment]
@karolis-adapt not really. Piping postcss after styleguide is canonical gulp way to achieve your goal.
Postccss integration is unnecessary here.

**Comment by karolis-adapt** [comment]
@glebmachine wait a sec. I think I misunderstood your previous comment. Currently, I'm generating the styleguide using the built css files. If I use styleguide with postcss files (not piping anything anywhere except for the read stream I guess) the error mentioned in the [original comment](https://github.com/SC5/sc5-styleguide/issues/825#issuecomment-157041322) gets logged out. Not using gulp, just cli interface of sc5 styleguide.

**Comment by glebmachine** [comment]
@karolis-adapt Ou, i missunderstood you too.
It's may possible to implement after this issue https://github.com/SC5/sc5-styleguide/issues/767

You still have option to create custom gulp postcss task, with full postprocessing chain btw.

**Comment by varya** [comment]
@faceg @karolis-adapt Could you try out this version, does it help? https://github.com/SC5/sc5-styleguide/releases/tag/0.3.41 It does not work with PostCSS plugins yet, but uses PostCSS parser for the source code. It may be enough to avoid StyleGuide fail on your project.

**Comment by karolis-adapt** [comment]
@varya the issue persists. [Updated the issue example repo to use the latest sc5-styleguide](https://github.com/karolis-adapt/sc5-postcss-testcase).

**Comment by faceleg** [comment]
Wow I look away for a second and this blows up.

Thanks for continuing to work towards this!

**Comment by stephenway** [comment]
Has anyone gotten this to work?
